staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/aggregator.go

```diff
@@ -59,12 +59,24 @@ func mergeSpec(dest, source *spec.Swagger) {
 		if dest.Definitions == nil {
 			dest.Definitions = spec.Definitions{}
 		}
-		dest.Definitions[k] = v
+		// We don't allow CRDs to have name conflict, but currently there is one corner
+		// case that a CRD can define apiextensions.apis.pkg.apiextensions-apiserver.k8s.io
+		// customresourcedefinitions, which causes definition conflict against
+		// apiextensions-apiserver's static spec. We skip the definition if it's already there
+		if _, found := dest.Definitions[k]; !found {
+			dest.Definitions[k] = v
```

Comment:
this is not the right place for this check. In general we allow those conflicts (and the merge algorithm deconflicts by renaming).

Reply:
Having said that, we might get defintions which are unused by paths (due to the change below). Somewhere we had a pruning step for that. Not sure this is effective though in the current plumbing for this situation.